### PR TITLE
Add explicit disclosure documentation to Ledger API services [DPP-1237]

### PR DESCRIPTION
**Pull request by tudor-da**
This PR adds a docs section describing the explicit disclosure feature as experimental starting from Canton 2.6 release. 

**Note**: The documentation is only targeting the Ledger API usage. Client support (e.g. Java bindings, Daml script etc.) implementation and documentation will be added later, prior to marking the feature GA.

### CODE REVIEW

File: docs/source/app-dev/submission-disclosed-contracts.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+during command interpretation.
+
+This means that usage of a disclosed contract effectively bypasses the visibility restriction of the submitting party's over the
+respective contract. However, the authorization restrictions of the Daml model still apply: the submitted command still needs to be well authorized (i.e. the actors
+need to be properly authorized to execute the action - as described in :ref:`Privacy Through Authorization <da-model-privacy-authorization>`).
+
+If we return to the example depicted in :ref:`Divulgence: When Non-Stakeholders See Contracts <da-model-divulgence>`,
+**ShowIou** is an auxiliary contract used by Alice to fetch the `IOU` under the Painter's
+projection in order to divulge it to him so that the Painter can use it when accepting the `CounterOffer`.
+This auxiliary step can be omitted if instead Alice chooses to share the `IOU`
```
Comment (simonmaxen-da) [inline docs/source/app-dev/submission-disclosed-contracts.rst:37]: May be worth mentioning that this sharing is done off-ledger currently.

File: docs/source/app-dev/services.rst [hunk 2]
```diff
@@ -92,6 +92,15 @@ The command submission service deduplicates submitted commands based on their :r
 
 For details on how to use command deduplication, see the :doc:`Command Deduplication Guide <command-deduplication>`.
 
+.. _command-submission-disclosed-contracts:
+
+(Experimental) Submitting commands with disclosed contracts
+-----------------------------------------------------------
+
+Starting with Canton 2.6, Ledger API clients can submit commands with attached
+disclosed contracts for bypassing the contract's visibility restriction to stakeholders. For more details,
```
Comment (simonmaxen-da) [inline docs/source/app-dev/services.rst:101]:
```suggestion
disclosed contracts bypassing the contract's visibility restriction to stakeholders. For more details,
```

File: docs/source/app-dev/submission-disclosed-contracts.rst (hunk 1 above)
Comment (simonmaxen-da) [reply docs/source/app-dev/submission-disclosed-contracts.rst:37]: I see this is covered below (link the word `share` to `How does the submitter get the disclosed contract details`) ?

**Review by simonmaxen-da — APPROVED** [review]

File: docs/source/app-dev/submission-disclosed-contracts.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
+.. Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+.. _submission-disclosed-contracts:
+
+(Experimental) Submitting commands with disclosed contracts
+###########################################################
+
+As described in :ref:`Divulgence: When Non-Stakeholders See Contracts <da-model-divulgence>`,
```
Comment (mziolekda) [inline docs/source/app-dev/submission-disclosed-contracts.rst:9]: What I would add here is a start on a positive, a preamble that states that with the version of 2.6 the old conundrum of delegating read rights of a contract **C** to a non-stakeholder party **P** has been solved elegantly by inclusion of the disclosed contract submission. Only then would I proceed to explain what was wrong with the previous two methods.

File: docs/source/app-dev/services.rst (hunk 2 above)
Comment (tudor-da) [reply docs/source/app-dev/services.rst:101]: 👍 Rephrased

File: docs/source/app-dev/submission-disclosed-contracts.rst (hunk 1 above)
Comment (tudor-da) [reply docs/source/app-dev/submission-disclosed-contracts.rst]: Good point. Linked

File: docs/source/app-dev/explicit-contract-disclosure.rst [hunk 4]
```diff
@@ -0,0 +1,94 @@
+.. Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+.. _explicit-contract-disclosure:
+
+(Experimental) Explicit contract disclosure
+###########################################
+
+Starting with Canton 2.6, explicit contract disclosure is introduced as an (experimental) feature
```
Comment (tudor-da) [inline docs/source/app-dev/explicit-contract-disclosure.rst:9]: Throughout the doc, I've referred to the artifact as Canton, even though we are in the Daml SDK docs. Double checking that this is fine and we don't need to consider the feature in a more general form, right?

File: docs/source/app-dev/explicit-contract-disclosure.rst [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+that allows delegating contract read rights to non-stakeholder parties in an optimal manner.
+Until now, read delegation was possible as well through the following mechanisms:
```
Comment (meiersi-da) [inline docs/source/app-dev/explicit-contract-disclosure.rst:11]: The way this is written a reader sort of needs to understand these two features, which we don't want them to understand at all, as they are sub-optimal.

I'd suggest to turn this around as follows: introduce a good example for off-ledger disclosure (a good one is IMHO reference data) and then explain how off-ledger disclosure is used to solve that. After explaining the desired solution, explain the existing alternatives and why they are not recommended.

Here are some notes form a recent conversation I've had about reference data for the exchange to serve a price-feed:

```
template PriceDatum with
    publisher : Party
    symbol : Text
    price : Decimal
    asOf : Time
  where
    signatory publisher

    nonconsuming choice PriceDatum_Read : PriceDatum
      with reader : Party
      controller reader
      do return this
```

and explain the topology where the exchange is connected to the domain on which their customers want to use the price feed.

If the exchange wants to bill customers based on the amount of reads they do then they can use:
```
template PriceDatum with
    publisher : Party
    symbol : Text
    price : Decimal
    asOf : Time
  where
    signatory publisher

    nonconsuming choice PriceDatum_Read : PriceDatum
      with reader : Party
      controller reader
      do return this
```

and use an off-ledger flow to bill the customer based on the number of validated tx nodes for `PriceDatum_Read`.

(this example requires further working out -- submitting now for consideration, as I need to head off)

**Review by meiersi-da — COMMENTED** [review]
partial review -- will continue tomorrow

**Comment by tudor-da** [comment]
Superseeded by https://github.com/digital-asset/daml/pull/16119

**Pull request closed without merging**